skaffold/src/main/kotlin/com/google/container/tools/skaffold/run/BaseSkaffoldSettingsEditor.kt

```diff
@@ -37,27 +38,46 @@ import javax.swing.JComponent
  * @param editorTitle title for the settings editor
  * @param helperText additional helper text for the settings editor
  */
-open class BaseSkaffoldSettingsEditor(val editorTitle: String, val helperText: String = "") :
-    SettingsEditor<AbstractSkaffoldRunConfiguration>() {
+open class BaseSkaffoldSettingsEditor<C : AbstractSkaffoldRunConfiguration>(
+    val editorTitle: String,
+    val helperText: String = ""
+) :
+    SettingsEditor<C>() {
 
     @VisibleForTesting
     val skaffoldFilesComboBox = SkaffoldFilesComboBox()
 
-    val basePanel = panel {
-        row {
-            label(helperText, 0, UIUtil.ComponentStyle.SMALL)
-        }
+    protected lateinit var basePanel: JPanel
+
+    private val extensionComponents: MutableMap<String, JComponent> = mutableMapOf()
 
-        row(message("skaffold.configuration.label")) { skaffoldFilesComboBox(grow) }
+    /**
+     * Registers additional custom components for Skaffold configuration UI.
+     * @param newExtensionComponents extension components mapped to their label text
+     */
+    protected fun addExtensionComponents(newExtensionComponents: Map<String, JComponent>) {
+        extensionComponents.putAll(newExtensionComponents)
     }
 
     override fun createEditor(): JComponent {
+        basePanel = panel {
+            row {
+                label(helperText, 0, UIUtil.ComponentStyle.SMALL)
+            }
+
+            row(message("skaffold.configuration.label")) { skaffoldFilesComboBox(grow) }
+
+            extensionComponents.forEach {
+                row(it.key) { it.value(grow) }
+            }
+        }
+
         basePanel.border = IdeaTitledBorder(editorTitle, 0, Insets(0, 0, 0, 0))
 
         return basePanel
     }
 
-    override fun applyEditorTo(runConfig: AbstractSkaffoldRunConfiguration) {
+    override fun applyEditorTo(runConfig: C) {
```

Comment:
why type `C`?

Reply:
`C[onfigurationType]`, to simplify child editors. You mean `C` is confusing instead of typical `T`?

Reply:
Ah its a generic type. I think T would be more standard, but I don't feel strongly

Reply:
According to Google Java style (http://google.github.io/styleguide/javaguide.html#s5-naming), naming either allows single letter (no strict recommendation on which exact letter though), or type name followed by a `T`, i.e. `RunConfigurationT`. I think both are fine, second being a bit more clear, but I think most of the developers already got used to `T`, `E` etc. We can still switch to `T` for maximum compatibility.

Reply:
ah. ok cool. Yeah Ive never seen the `RunConfigurationT` style before - I don't really like it either myself.